lib/view_component/base.rb

```diff
@@ -31,6 +31,10 @@ class Base < ActionView::Base
 
     attr_accessor :__vc_original_view_context
 
+    # Lock required to be obtained before compiling the component
+    class_attribute :lock
+    self.lock = Monitor.new
```

Comment:
```suggestion
    class_attribute :__vc_compiler_lock
    self.__vc_compiler_lock = Monitor.new
```

Can we follow the "internal attribute" naming convention here?

Reply:
Alternative thought, would it be possible to put the monitor in the compiler instead of on this specific component instance? Each component should have a single compiler instance, as defined on https://github.com/github/view_component/pull/1161/files#diff-b69c75ffd85596e0b5fca1327059ecf3b2930f3f77004c78a268fa85e8ec8d9eR412

Reply:
I will look into it. Thanks a lot for the suggestions!

Reply:
Awesome idea! This simplified things a bit, since I didn't have to take care of ensuring that each subcass of `ViewComponent::Base` had its own lock instance. Didn't realise that each component class had its own compiler instance!